firebase_functions/functions/src/migration.ts

```diff
@@ -0,0 +1,150 @@
+import { Client, ClientBuilder } from '@iota/client';
+import * as admin from 'firebase-admin';
+import * as crypto from 'crypto';
+
+export const migration = async (from: number, to: number) => {
+  await deleteUnusedDevices();
+  await deleteUnusedUsers(from, to);
+  await updateDevices(from, to);
+  await updateUser(from, to);
+};
+
+const deleteUnusedDevices = async () => {
+  const deviceRef = admin.firestore().collection('devices');
+  await deviceRef.get().then(async (querySnapshot) => {
+    const docs = querySnapshot.docs;
+    console.log("Docs length: ", docs.length)
+    for (let docIndex = 50; docIndex < 174; docIndex++ ) {
+      const device = docs[docIndex];
+      console.log("Device: ", device.id)
+      const data = await admin.firestore().collection('devices').doc(device.id).collection('data').limit(1).get();
+      if (data.size === 0) {
+        console.log("------> Deleting device: ", device.id)
+        await admin.firestore().collection('devices').doc(device.id).delete();
+        await admin.firestore().collection('deviceList').doc(device.id).delete();
+      }
+    }
+  });
+};
+
+const deleteUnusedUsers = async (from: number, to: number) => {
+    const userRef = admin.firestore().collection('users');
+    await userRef.get().then(async (querySnapshot) => {
+      const docs = querySnapshot.docs;
+      console.log("Docs length: ", docs.length)
+      for (let docIndex = from; docIndex < to; docIndex++ ) {
+        const device = docs[docIndex];
```

Comment:
I think, you can safely rename `device` into `user` to avoid confusions

Reply:
Done